libgalois/src/analytics/louvain_clustering/louvain_clustering.cpp

```diff
@@ -492,10 +492,36 @@ struct LouvainClusteringImplementation
   }
 };
 
+template <typename EdgeWeightType>
+static katana::Result<void>
+AddDefaultEdgeWeight(
+    katana::PropertyGraph* pg, const std::string& edge_weight_property_name) {
+  using EdgeData = std::tuple<EdgeWeightType>;
+
+  if (auto res = katana::analytics::ConstructEdgeProperties<EdgeData>(
+          pg, {edge_weight_property_name});
+      !res) {
+    return res.error();
+  }
+
+  auto typed_graph =
+      KATANA_CHECKED((katana::TypedPropertyGraph<std::tuple<>, EdgeData>::Make(
+          pg, {}, {edge_weight_property_name})));
+  katana::do_all(
+      katana::iterate(typed_graph),
+      [&](auto n) {
+        for (auto ii : typed_graph.edges(n)) {
+          typed_graph.template GetEdgeData<EdgeWeightType>(ii) = 1;
+        }
+      },
+      katana::steal(), katana::loopname("InitEdgeWeight"));
```

Comment:
Why not iterate over the edges instead of the nodes? You can do that with `typed_graph.edges()` I think. Then you wouldn't need `steal` and the code would be a bit simpler, *and* perhaps a little tiny bit faster.

Reply:
There is no iterate over edges TypedPropertyGraph. I don't see one.

Reply:
It's on GraphTopology: https://github.com/KatanaGraph/katana/blob/f21fd0c5ccda6fd335805df36584d1b9368cdf2e/libgalois/include/katana/GraphTopology.h#L133